### PR TITLE
Add config file strucure

### PR DESCRIPTION
**Pull request by letsgogeeky**

### CODE REVIEW

File: include/configFile/HttpContext.hpp [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  std::string _fastcgi_passValue;
+  std::string _fastcgi_indexValue;
+  std::string _includeValue;
+  std::string _uwsgi_passValue;
+  std::string _scgi_passValue;
+  std::string _mp4Value;
+  std::string _mp4_buffer_sizeValue;
+  std::string _mp4_max_buffer_sizeValue;
+  std::string _sub_filterValue;
+  std::string _sub_filter_onceValue;
```
Comment (letsgogeeky) [inline include/configFile/HttpContext.hpp:52]: I don't think we need this

File: include/configFile/Location.hpp [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  std::string _urlValue;
+  std::string _rootValue;
+  std::string _proxy_passValue;
+  std::string _aliasValue;
+  std::string _try_filesValue;
+  std::string _indexValue;
+  std::string _error_pageValue;
+  std::string _rewriteValue;
+  std::string _returnValue;
```
Comment (letsgogeeky) [inline include/configFile/Location.hpp:18]: this as well

File: include/configFile/Location.hpp [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  std::string _limit_reqValue;
+  std::string _limit_connValue;
+  std::string _proxy_set_headerValue;
+  std::string _proxy_redirectValue;
+  std::string _fastcgi_passValue;
+  std::string _uwsgi_passValue;
+  std::string _scgi_passValue;
+  std::string _ContextsValue;
+  std::string _ifValue;
+  std::string _limit_exceptValue;
```
Comment (letsgogeeky) [inline include/configFile/Location.hpp:33]: we won't need this as well

**Review by letsgogeeky — COMMENTED** [review]
Inside the location there can be a CGI config that looks like that:
```
cgi .py /usr/bin/python
```

File: include/configFile/Location.hpp (hunk 3 above)
Comment (fritz-gabler) [reply include/configFile/Location.hpp:33]: is removed

File: include/configFile/Location.hpp (hunk 2 above)
Comment (fritz-gabler) [reply include/configFile/Location.hpp:18]: also removed

File: include/configFile/HttpContext.hpp [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+class HttpContext {
+ public:
+  HttpContext();
+
+  unsigned int _numberOfServer;
+  std::vector<ServerContext> _serverContext;
+  Location _locationContext;
+
+  std::string _geoipCountry;
```
[the remark this answers is not in the document]
Comment (fritz-gabler) [reply include/configFile/HttpContext.hpp:17]: std::string _geoipCountry; shall we remove that? but the comment does not include the vector and the Location,  right?

**Pull request closed without merging**